<launch>
  <!-- clock -->
  <param name ="/use_sim_time" value="true"/>

  <!-- transforms -->
  <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0 0 0 0 0 0 1  /map /odom 10" />
  <node pkg="tf" type="static_transform_publisher" name="odom_to_base" args="0 0 0 0 0 0 1  /odom /base_link 10" />
  <node pkg="tf" type="static_transform_publisher" name="gps_init_to_baselink"  args="0 0 0 0 0 0 1 /base_link /gps  10" />
  <node pkg="tf" type="static_transform_publisher" name="base_to_cam"  args="0.343 0.079 0.820 -0.002 0.100 -0.004 0.995 /base_link /zed_left_camera_frame 10" />
  <node pkg="tf" type="static_transform_publisher" name="base_to_velodyne"  args="0.286 0.025 0.920 -0.010 0.002 -0.045 0.999 /base_link /velodyne 10" />

  <!-- rviz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find vineslam_ros)/param/config.rviz" required="false"/>

  <!-- agrob_map_transform gps node -->
  <node pkg="agrob_map_transform" type="agrob_gnss_transform_node" name="agrob_gnss_transform_node" clear_params="true" output="screen">
    <rosparam command="load" file="$(find agrob_map_transform)/params/navsat_transform_template.yaml" />
<!--    <rosparam command="load" file="$(find agrob_map_transform)/params/vineslam_configs.yaml" />-->

      <remap from="imu/data" to="/Dummy/IMU"/>
      <remap from="odometry/filtered" to="/vineslam/odom"/>
      <remap from="gps/fix" to="/fix"/>
  </node>

  <!-- image republish -->
  <node name="republish1" type="republish" pkg="image_transport" output="screen"
        args="compressed in:=/zed/zed_node/left_raw/image_raw_color out:=/zed/zed_node/left/image" />

  <!-- slam node -->
  <arg name="bagfile_name" default="/home/andreaguiar/Documents/rosbags/estufa/tomates_2020-08-06-11-23-35.bag"/>
  <node pkg="vineslam_ros" type="slam_node" name="slam_node" output="screen">
    <rosparam command="load" subst_value="True" file="$(find vineslam_ros)/param/setup.yaml"/>
  </node>

  <!-- Coral edgetpu trunk detection node -->
  <node pkg="detector" type="detector_node" name="detector" output="screen">
    <param name="image_left"  value="/zed/zed_node/left/image"/>
    <param name="image_depth" value="/zed/zed_node/depth/depth_registered"/>

    <param name="model_path"  value="$(find edgetpu_cpp)/model/output_tflite_graph_edgetpu.tflite"/>
    <param name="labels_path" value="$(find edgetpu_cpp)/model/labels.txt"/>
  </node>

</launch>
